service_discovery/dto/service_definition.go

```diff
@@ -17,6 +17,43 @@
 
 package dto
 
+import "encoding/json"
+
+// ServiceDefinition interface is interface for all service definition types
 type ServiceDefinition interface {
 	GetLocation() Location
 }
+
+// UnsupportedServiceDefinition represents unknown or unsupported service definition returned by deserializer
+type UnsupportedServiceDefinition struct {
+}
+
+// GetLocation always panics on unsupported service types
+func (UnsupportedServiceDefinition) GetLocation() Location {
+	//no location available - should never be called
+	panic("not supported")
+}
+
+var _ ServiceDefinition = UnsupportedServiceDefinition{}
+
+// ServiceDefinitionUnserializer defines function to register for concrete service definition
+type ServiceDefinitionUnserializer func(*json.RawMessage) (ServiceDefinition, error)
+
+// service definition unserializer registry
+var serviceDefinitionMap = make(map[string]ServiceDefinitionUnserializer, 10)
```

Comment:
Why do we need to preallocate map with a size of 10 here?

Reply:
No idea. It's not my code. But it can be safely removed anyway.